### ISSUE TITLE
Auth error - Error: Failed to fetch when attempting to use Authorize feature in preview

### ISSUE THREAD

**Issue by sulphy** [42Crunch/vscode-openapi#97]
I'm attempting to use the Authorize feature for an endpiont that is protected by OAuth2.  The endpoints setup in the swagger spec are fine (tested in Postman).  

1. Open the Swagger spec in the preview pane
2. Click the 'Authorize' button
3. Enter client_Id and client_secret

For reference the security schema in the spec is:

```
    "components": {
        "securitySchemes": {
            "oAuthSample": {
                "type": "oauth2",
                "description": "",
                "flows": {
                    "clientCredentials": {
                        "tokenUrl": "{removed}",
                        "scopes": {
                            "read_test": "read your test",
                            "write_test": "modify test in your account"
                        }
                    }
                }
            }
        }
```

![2020-11-17 22_03_05-OpenAPI SwaggerUI preview - Untitled (Workspace) - Visual Studio Code](https://user-images.githubusercontent.com/25533844/99456172-dc7ffe00-2920-11eb-9866-5e5f7b9df233.png)

**Comment by klausbadelt** [comment]
This happens for us not only in this specific case - _all_ Swagger preview requests fail with `TypeError: Failed to fetch`. I suspect it's related to the webview's Content Security Policy -  the Developer Tools console shows `Refused to connect to 'http://localhost:4010/uploads' because it violates the document's Content Security Policy.`

<img width="498" alt="Screen Shot 2020-11-18 at 11 51 22 PM" src="https://user-images.githubusercontent.com/5615403/99598108-52f02f00-29f9-11eb-880e-7628ef56e238.png">

**Comment by ak1394** [comment]
@sulphy OAuth authorization is not currently supported for Try Out feature in Swagger UI preview. We might look into supporting it in the future though.

**Comment by ak1394** [comment]
@klausbadelt  could you confirm that you're seeing it with the latest (3.8.0) version of the extension? Could you share your OpenAPI (or a snippet of it) which exhibits the problem?

**Comment by klausbadelt** [comment]
@ak1394 Hi Anton - same issue with 3.8.0. Tried different devbox also (this time Windows), same result.

Here's a minimal non-working OpenAPI:
```yaml
openapi: 3.0.2
info:
  title: Filmhub Backend API
  version: '2.0'
  description: 'Filmhub''s backend server API'
servers:
  - url: '*** redacted ***'
    description: Rails API Mock server
security:
  - jwt: []
components:
  securitySchemes:
    JWT:
      type: http
      description: JWT Authentication
      scheme: bearer
      bearerFormat: JWT
paths:
  /titles:
    get:
      responses:
        '200':
          description: OK
```

**Comment by ak1394** [comment]
@klausbadelt Hi Klaus! Thanks for the sample, I tried it and don't seem to have any issues with it. 

Cold you try checking the DevTools for errors? The instructions on how to bring it up are here: https://code.visualstudio.com/api/extension-guides/webview#inspecting-and-debugging-webviews 

One thing to notice, is that for Try Out to work you must have properly configured CORS on your server.

**Comment by klausbadelt** [comment]
@ak1394 I had posted the DevTools output [above](https://github.com/42Crunch/vscode-openapi/issues/97#issuecomment-730008597). Here's a screenshot too. Hope this helps.

<img width="1156" alt="Screen Shot 2020-12-01 at 7 09 38 PM" src="https://user-images.githubusercontent.com/5615403/100779683-06e5b700-3409-11eb-83c4-9f48e4d200d8.png">

**Comment by klausbadelt** [comment]
I'm not sure what you mean with "properly configured CORS on your server" - the request is from within VS Code's WebView, my sever doesn't even get hit?

**Comment by ak1394** [comment]
Thanks for the screenshot! It certainly helps, see on the first line of the trace it says that "connect-src" was not set, although the last version (3.8) has it set: https://github.com/42Crunch/vscode-openapi/blob/341689770e2cf6bc5fea2d9b6502b4a21cc7e6c9/src/preview.ts#L151 

I'll check if it could be some sort of caching issue in VS Code, but in the meantime could you try uninstalling and re-installing the extension?

Regarding the CORS remark, I wasn't sure about the timing and thought perhaps you had your initial issue with v3.7 which didn't have connect-src fix, and the subsequent failures happened  with v3.8 and were caused by CORS on the server. However the screenshot shows it's not the case.

**Comment by bluppfisk** [comment]
Following

**Comment by klausbadelt** [comment]
Hi @bluppfisk sorry for the delay.

Upgraded to current 3.10.0. Seems to work now, and I have to figure out the CORS settings.

```
[Embedded Page] Access to fetch at 'https://***.stoplight.io/***/***/api_backend/titles/12345/videos' from origin 'vscode-webview://68942489-a3c0-40cd-b592-99ba1f41d2ac' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
webviewElement.ts:90 [Embedded Page] TypeError: Failed to fetch
```

**Comment by ak1394** [comment]
@klausbadelt So looks like the problem is gone in the current version,  @bluppfisk feel free to open a new issue if needed.
